### ISSUE TITLE
Isolating styles

### ISSUE THREAD

**Issue by mayank99** [iTwin/iTwinUI#1033]
Current goal: v1 and v2 should be able to coexist on a page, to allow some form of incremental migration for apps.

Future goal: all versions should be fully scoped and independent of each other. (aka [CSS modules](https://github.com/css-modules/css-modules)).

**Comment by mayank99** [comment]
@GintV Since you expressed interest in using iTwinUI v2 alongside v1 (via iTwin.js 3.x), I think this issue will be of interest to you. Please let us know if you're interested in testing.

I have come up with a way for v1 and v2 to coexist such that you can start using v2 in isolated parts of your page, while still using v1 for the rest of the page. Here is a sandbox showing it in action: https://stackblitz.com/edit/vitejs-vite-7qrmeg?file=src%2FApp.jsx

In a real app, the approach would need some adjustment, but the idea is the same. You may need to use nested (p)npm overrides to make sure that correct versions are used.

We are currently testing this with another team. If you want to test it in your app, then use these dev versions:
```
"@itwin/itwinui-css": "0.63.2-dev.1",
"@itwin/itwinui-react": "1.48.2-dev.2",
```
```
"@itwin/itwinui-css": "1.5.0-dev.2",
"@itwin/itwinui-react": "2.4.0-dev.3",
"@itwin/itwinui-variables": "2.0.0-dev.1",
```

When using [pnpm overrides](https://pnpm.io/package_json#pnpmoverrides), it might look something like this (verify your lockfile after installing):
```
"@itwin/itwinui-css@0": "0.63.2-dev.1",
"@itwin/itwinui-css@1": "1.5.0-dev.2",
"@itwin/itwinui-react@1": "1.48.2-dev.2",
"@itwin/itwinui-react@2": "2.4.0-dev.3",
```

After that, your v1 code can remain unchanged, and you can created scoped roots for v2 by wrapping them in `ThemeProvider`.

**Comment by mayank99** [comment]
This is now released in `@itwin/itwinui-react@2.5.0` and `@itwin/itwinui-react@1.48.2`.

How it will work:
1. Packages that contain v2 components will upgrade to `@itwin/itwinui-react@2.5.0` and wrap their entrypoint with `<ThemeProvider theme='inherit'>...</ThemeProvider>`. See [example PR](https://github.com/iTwin/synchronization-report-react/pull/61).
2. Apps will upgrade to the new version of the package, and then use [(p)npm overrides](https://pnpm.io/package_json#pnpmoverrides) similar to this:
 	```
 	"@itwin/itwinui-css@0": "0.63.2",
 	"@itwin/itwinui-css@1": "1.6.0",
 	"@itwin/itwinui-react@1": "1.48.2",
 	"@itwin/itwinui-react@2": "2.5.0",
 	``` 
   This ensures that the correct iTwinUI versions are used within nested dependencies and that there are no duplicates.
